deepinterpolation/generator_collection.py

```diff
@@ -897,112 +918,134 @@ class OphysGenerator(SequentialGenerator):
     def __init__(self, json_path):
         "Initialization"
         super().__init__(json_path)
+        self._gpu_available = tf.test.is_gpu_available()
+        self._movie_data = None
+        if self._gpu_available:
+            self._cached_index = None
+            self._data_tensor = None
+            self.batch_indices = None
+            self.input_indices = None
 
         # For backward compatibility
-        if "train_path" in self.json_data.keys():
-            self.raw_data_file = self.json_data["train_path"]
-        else:
-            self.raw_data_file = self.json_data["movie_path"]
-
+        self.cache_data = self.json_data.get("cache_data", False)
+        self.raw_data_file = self.json_data.get(
+            "train_path", self.json_data.get("movie_path"))
+        self.gpu_cache_full = self.json_data.get("gpu_cache_full", False)
```

Comment:
A user with only a cpu could pass `gpu_cache_full` = True. Then you log "Caching full movie onto GPU" which isn't true. You need to make sure that the user has a gpu if `gpu_cache_full` = True. You need to validate here that there actually is a gpu if True is passed.